mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/dependencies/DependencyResolver.java

```diff
@@ -0,0 +1,71 @@
+package org.arquillian.smart.testing.mvn.ext.dependencies;
+
+import java.util.Map;
+import java.util.Optional;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Plugin;
+import org.arquillian.smart.testing.Configuration;
+import org.arquillian.smart.testing.mvn.ext.ApplicablePlugins;
+
+public class DependencyResolver {
+
+    private final Configuration configuration;
+
+    public DependencyResolver(Configuration configuration) {
+        this.configuration = configuration;
+    }
+
+    public void addRequiredDependencies(Model model) {
+        addStrategies(model);
+        addSurefireApiDependency(model);
+    }
+
+    private void addStrategies(Model model) {
+        final String[] strategies = configuration.getStrategies();
+        final StrategyDependencyResolver strategyDependencyResolver = new StrategyDependencyResolver();
+        model.addDependency(smartTestingProviderDependency());
+        final Map<String, Dependency> dependencies = strategyDependencyResolver.resolveDependencies();
+        for (final String strategy : strategies) {
+            final Dependency dependency = dependencies.get(strategy);
+            model.addDependency(dependency);
+        }
+    }
+
+    public void addAsPluginDependency(Plugin plugin) {
+        plugin.addDependency(smartTestingProviderDependency());
+    }
+
+    private void addSurefireApiDependency(Model model) {
+        boolean alreadyContains = model.getDependencies()
+            .stream()
+            .anyMatch(dep -> "org.apache.maven.surefire".equals(dep.getGroupId()) && "surefire-api".equals(
+                dep.getArtifactId()));
+        if (!alreadyContains) {
+            final Optional<Plugin> surefirePlugin = model.getBuild().getPlugins().stream()
+                .filter(plugin -> ApplicablePlugins.SUREFIRE.hasSameArtifactId(plugin.getArtifactId()))
+                .findFirst();
+
+            surefirePlugin.ifPresent(plugin -> model.addDependency(
+                getSurefireApiDependency(Version.from(plugin.getVersion().trim()).toString())));
+        }
+    }
+
+    private Dependency getSurefireApiDependency(String version) {
+        final Dependency dependency = new Dependency();
+        dependency.setGroupId("org.apache.maven.surefire");
+        dependency.setArtifactId("surefire-api");
+        dependency.setVersion(version);
+        dependency.setScope("runtime");
+        return dependency;
+    }
```

Comment:
how about the separated class?
```java
public class SurefireApiDependency extends Dependency {

    private static final String GROUP_ID = "org.apache.maven.surefire";
    private static final String ARTIFACT_ID = "surefire-api";

    public SurefireApiDependency(String version) {
        setGroupId(GROUP_ID);
        setArtifactId(ARTIFACT_ID);
        setVersion(version);
        setScope("runtime");
    }

    public static boolean matches(String groupId, String artifactId) {
        return GROUP_ID.equals(groupId) && ARTIFACT_ID.equals(artifactId);
    }
}
```
the method `matches` could be used here: https://github.com/arquillian/smart-testing/pull/88/files#diff-e5427bc921ef9faf2177273978bd12feR42